### ISSUE TITLE
Closing an Admin Interface Listbox Contained in an Edit Modal Attempts to Close the Modal

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2425]
In the Admin Interface, a lot of Editing happens in a modal on the side of the screen. If there is a Listbox contained in that modal, attempting to close the Listbox instead attempts to close the whole modal, which causes confusion and could potentially lead to accidental deletion of edits.

## Expected Behavior
Closing an expanded Listbox should close the Listbox and place focus back on the button that opened that opened it.

## Current Behavior
Closing certain Listboxes in the Admin Interface opens an unrelated popup modal, causing confusion.

## Steps to Reproduce
1. Navigate to Admin Interface (via Enter Admin Mode)
2. Navigate to an editable project
3. Navigate to the Layout page
4. Edit the Calls to Action, and activate the "Add Button to Left Side" button
5. Navigate to and activate the "type" Listbox
6. Press the arrow down key to change the selected option
7. Press escape to close the Listbox
8. Notice how a new "Are you Sure" modal pops up (also note that "Are You Sure" is not announced, but this will be its own issue).
9. Note how continually pressing escape continues to close and reopen this modal

## Testing Environment
NVDA 2019.2.1
ESR Firefox 68.2.0 esr

**Comment by dananjohnson** [comment]
Thanks for documenting this for us @Santoz2. I'm actually seeing _four different issues_ when I review this:

1. Pushing ESC with the listbox active triggers the drawer's keyup listener. I'm seeing this in Firefox and Safari, but not Chrome.
2. The "Are you sure?" modal isn't announced.
3. Pushing ESC in the "Are you sure?" modal triggers the drawer's keyup listener.
4. The color of the listbox is black in Firefox. This likely can be traced to trying to set a color for the options, since FF is unique in inheriting this color from the `<select>`, and would otherwise appear white in this context.
<img width="520" alt="Screen Shot 2019-11-07 at 2 25 40 PM" src="https://user-images.githubusercontent.com/12454203/68433342-214c5300-016b-11ea-93ff-ce6922086b61.png">

**Comment by dananjohnson** [comment]
I'm going to break these up and track them as separate issues.

**Comment by dananjohnson** [comment]
No. 2 is tracked at #2430 
No. 4 is tracked at #2429

**Comment by Santoz2** [comment]
Thank you for breaking these up!

**Comment by zdavis** [comment]
@dananjohnson you need anything from me to close this one out?

**Comment by dananjohnson** [comment]
@zdavis as far as I can tell, this issue was in mind when the Drawer component was initially built, but the attempted solution doesn't actually work. Specifically, `<Drawer />` tries to use the Context API to pause keyboard events when a Dialog component is mounted. However, the the Dialog is higher up the tree than Drawer, so it can't actually be a consumer of the Drawer's context. Before jumping back into this, I wanted to check with you about how best to resolve this. It looks like the context provider would to be near the very top of the tree if it's going to be consumed by `<Dialog />`. 

Sidenote: I started by updating to the new Context API here, since the old API is now deprecated.

**Comment by dananjohnson** [comment]
@zdavis it's been awhile, so let me take a look again. I should be able to solve it.

**Comment by dananjohnson** [comment]
I can no longer reproduce this issue. I'm not sure exactly which commit resolved it, but we've made a number of improvements to focus management in drawers since this issue was originally created, and listbox interactions now behave as expected. I'm going to go ahead and close this issue as resolved.
